### ISSUE TITLE
Page hierarchy explorer doesn't show element properties when they're not named "children"

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#3117]
### Search keywords

children

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: (required)

Steps:
1. Create a custom component:
    ```tsx
    import * as React from 'react';

    import { createComponent } from '@mui/toolpad/browser';

    export interface MyComponentProps {
      notChildren: React.ReactNode;
    }

    function MyComponent({ notChildren }: MyComponentProps) {
      return <div>hello:{notChildren}</div>;
    }

    export default createComponent(MyComponent, {
      argTypes: {
        notChildren: {
          type: 'element',
          control: { type: 'slots' },
        },
      },
    });
    ```
2. Drag into the page, and drag a button component inside of the "Drop component here" area.
3. Check the hierarchy explorer

### Current behavior

The button doesn't appear in the hierarchy explorer

### Expected behavior

The button appears in the hierarchy explorer. Just like it does when you replace `notChildren` the custom component with `children`

### Context

Noticed [here](https://github.com/mui/mui-toolpad/pull/3100). It looks like to solve this correctly, we need to signal in our dom when something is an element/template, independently from the components. I'm trying to introduce an `{ $$element: {...} }` bindable property in the app dom and replace the `elm.children` property with a "children" property under `elm.props.children` that has that `$$element` form. 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by apedroferreira** [comment]
Yeah this is a bit messy, there were always a few issues like this when supporting custom components with slots, I guess there might still be some.
This might be a regression too as it probably worked before, it should be covered by tests.

**Comment by JerryWu1234** [comment]
is there  only replacement of `elm.props.children` to `elm.props.$$element`. right?

**Comment by JerryWu1234** [comment]
@Janpot there aren't a lot of bugs in the issue list that I can fix sometime.
So I want to ask for give me some requirements that not include in the issue list.
